runtime/compiler/codegen/J9AheadOfTimeCompile.cpp

```diff
@@ -563,6 +563,23 @@ J9::AheadOfTimeCompile::initializeCommonAOTRelocationHeader(TR::IteratedExternal
          }
          break;
 
+      case TR_BlockFrequency:
+         {
+         TR_RelocationRecordBlockFrequency *blockFrequencyRecord = reinterpret_cast<TR_RelocationRecordBlockFrequency *>(reloRecord);
+         TR_PersistentProfileInfo *profileInfo = TR::comp()->getRecompilationInfo()->getProfileInfo();
+
+         TR_ASSERT(NULL != profileInfo, "PersistentProfileInfo not found when creating relocation record for block frequency\n");
+
+         TR_BlockFrequencyInfo *blockInfo = profileInfo->getBlockFrequencyInfo();
+
+         TR_ASSERT(NULL != blockInfo, "BlockFrequencyInfo not found when creating relocation record for block frequency\n");
+
+         uintptr_t frequencyArrayBase = reinterpret_cast<uintptr_t>(blockInfo->getFrequencyArrayBase());
+         uintptr_t frequencyPtr = reinterpret_cast<uintptr_t>(relocation->getTargetAddress());
+
+         blockFrequencyRecord->setFrequencyOffset(reloTarget, frequencyPtr - frequencyArrayBase);
+         }
```

Comment:
Missing a `break;` here.

Also you should add code to dump the relo data further below in `J9::AheadOfTimeCompile::dumpRelocationHeaderData`